DESIGN_SPEC.md

```diff
@@ -1,107 +1,258 @@
 
-## Problem
-- SQLAlchemy is new to the team
-- There is alot of ways to do it un-scalably, and we want to reduce the surface area
-of problems for the team for the easiest adoption
-  
+Before that, let's give a quick introduction on SQLAlchemy.
 
-## Potential Solutions
-- Creating a Thin Abstraction over SQLAlchemy for Queries
-- Creating a recipe guidebook in Confluence and/or this repo
 
-## Examples and Use Cases
+## SQLAlchemy: Introduction + Table Setup
+### About the Library
+SQLAlchemy is a library that allows us to work with databases using Python. There are two main user-facing portions to SQLAlchemy: Core and ORM. At KF, we will focus on using SQLAlchemy ORM.
 
 
-#### Selecting certain columns ("select" clause in SQL)
-The SQL example we are trying to do:
+### Explaining ORM
+ORM, or Object Relational Mapper, allows us to write SQL Queries using object oriented concepts. This is helpful because it allows us to query a database using any language, it allows us to use any database system the same way (no need to memorize the nuances between MySQL syntax versus Postgres), and it provides a lot of added functionality. SQL query strings are also vulnerable to attack, and ORMs can prevent some SQL injection attempts.
+
+### Definitions
+#### Primary and Foreign Keys
+A primary key is a column in a table whose values uniquely identify a row in the table, and a foreign key is a column that corresponds tot he values of the primary key in another table. Together, the primary and foreign keys define relationships across tables. This will be useful for knowing how to join tables later on.
+
+
+## Problem Setup
+Let's say we have a database that we use for pets of KF employees. We use this database to find pet birthdays, whose pets can come into the office, and the most popular breeds among KF employees.
+
+It has the employee's basic information, whether they have pets, information about any pets, if they have pet insurance through KF (made up, we do not over this, not even sure if this exists) and pets that KF allows into the office.
+
+We have 5 tables: employees, pets, pet_info, insurance, office_pets
+
+### Defining Our SQL Tables
+Let's define our tables. Some tables will have primary keys and foreign keys. 
+
+Here are our tables, primary and foreign keys:
+
+Database: kungfu_pets
+
+Table: employees
+Primary key: employee_id
+
+employee_id | employee_name  
+------------|----------------
+1			|	Endurance
+2			|	Tony
+3			|	Krishi
+4			|	Max M.
+5 			|	Reed C.
+
+Table: pets
+Primary key: pet_name
+Foreign key: employee_id
+
+pet_name    | employee_id   
+------------|----------------
+Intel		|	1
+Misa		|	1
+Java	   	|	1	
+Shadow		|	2
+Callie  	|	4
+Yoshi		|	4
+Deohgie		|	4
+
+Table: pet_info
```

Comment:
Strings should not be our primary keys. And in this case, if we have pet_name collisions, we would have a problem.